examples/shader/Shader.cpp

```diff
@@ -21,46 +21,58 @@ std::random_device rd;
 std::mt19937       rng(rd());
 } // namespace
 
+
+////////////////////////////////////////////////////////////
+// Base class for effects
+////////////////////////////////////////////////////////////
+struct Effect : public sf::Drawable
+{
+    virtual void update(float time, float x, float y) = 0;
+};
+
+
 ////////////////////////////////////////////////////////////
 // "Pixelate" fragment shader
 ////////////////////////////////////////////////////////////
 class Pixelate : public Effect
 {
 public:
-    Pixelate() : Effect("Pixelate")
+    static std::optional<Pixelate> tryLoad()
     {
-    }
+        auto texture = sf::Texture::loadFromFile("resources/background.jpg");
+        if (!texture.has_value())
+            return std::nullopt;
 
-    bool onLoad() override
-    {
-        // Load the texture and initialize the sprite
-        if (!(m_texture = sf::Texture::loadFromFile("resources/background.jpg")))
-            return false;
-        m_sprite.emplace(*m_texture);
+        auto shader = sf::Shader::loadFromFile("resources/pixelate.frag", sf::Shader::Type::Fragment);
+        if (!shader.has_value())
+            return std::nullopt;
 
-        // Load the shader
-        if (!(m_shader = sf::Shader::loadFromFile("resources/pixelate.frag", sf::Shader::Type::Fragment)))
-            return false;
-        m_shader->setUniform("texture", sf::Shader::CurrentTexture);
+        return Pixelate{std::move(*texture), std::move(*shader)};
+    }
 
-        return true;
+    explicit Pixelate(sf::Texture&& texture, sf::Shader&& shader) :
+    m_texture{std::move(texture)},
```

Comment:
```suggestion
    m_texture(std::move(texture)),
```
By convention we use `( ... )` in initializer lists.